backend/src/schedulers/lock-reports-scheduler.ts

```diff
@@ -0,0 +1,79 @@
+import { CronJob } from 'cron';
+import { config } from '../config';
+import { logger as log } from '../logger';
+import advisoryLock from 'advisory-lock';
+import prisma from '../v1/prisma/prisma-client';
+import { LocalDateTime, ZoneId, convert } from '@js-joda/core';
+
+try {
+  const mutex = advisoryLock(config.get('server:databaseUrl'))('lock_reports');
+  const crontime = config.get('server:schedulerLockReportCronTime');
+  const timezone = config.get('server:schedulerTimeZone');
+  const reportEditDurationInDays = config.get(
+    'server:reportEditDurationInDays',
+  );
+  const reportUnlockDurationInDays = config.get(
+    'server:reportUnlockDurationInDays',
+  );
+
+  const job = new CronJob(
+    crontime, // cronTime
+    async function () {
+      try {
+        const unlock = await mutex.tryLock();
+        if (unlock) {
+          log.info('Starting report locking Schedule Job.');
+
+          await prisma.pay_transparency_report.updateMany({
```

Comment:
wrap this in a transaction boundary

Reply:
This is a single independent write, there is no value now, but it does not hurt to wrap it with a transaction